packages/babel-generator/src/generators/methods.ts

```diff
@@ -204,3 +222,46 @@ function hasTypesOrComments(
     param.trailingComments?.length
   );
 }
+
+function _getFuncIdName(this: Printer, idNode: t.Node, parent: t.Node) {
```

Comment:
It seems to me here we are implementing something similar to https://github.com/babel/babel/tree/main/packages/babel-helper-function-name, which is also used in this decorator transform PR (https://github.com/babel/babel/pull/15122). Can we somehow unify them all?

Cases missed:
- assignment expression (`x = function() {}`)
- assignment pattern (`{ x = function() {} } = {}`)
- class private property (`#x = function() {}`)
- class accessor property (`accessor x = function() {}`)

Reply:
> Can we somehow unify them all?

I'm not sure it's worth unifying them, `babel-helper-function-name` uses something that belongs to `traverse`, and it's relying on `@babel/template`, and they can only have a small amount of code in common.

Also I accidentally found that `babel-helper-function-name` doesn't seem to maintain the source map very well, which is a bit bad, maybe we need to improve it in the future.

> Cases missed:

Amazing find!